plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -2020,32 +2020,23 @@ fu_genesys_usbhub_device_update_firmware(FuGenesysUsbhubDevice *self,
 					 GError **error)
 {
 	gboolean skip_erase = FALSE;
+	gsize bufsz = 0;
+	const guint8 *buf = NULL;
 	const guint start_addr = fu_firmware_get_addr(firmware);
-	const guint code_size = fu_firmware_get_size(firmware);
```

Comment:
As the result, the `code_size` is necessary for the firmware.

Reply:
> BTW, this also needs to replace 

I've added a fixup -- can you email me your test firmware please richard_at_hughsie_dot_com and I'll make sure we include the extra sub firmware types in the fuzzing target -- which will also catch things like this.

Reply:
Please try the binaries I sent.

Reply:
I've tried a few of the firmwares and they seem to have the same putput as before; i.e. the parsing is identical. Can you point me in the right direction please?

Reply:
```
    <size>0xd000</size>
    <data size="0x15000">.{..6....&quot;&quot;..r..</data>
    ```

    The firmware should only write the `size` of data (0xd000), not `data size`(0x15000).

Reply:
aha, right, got it; give me 10 mins.

Reply:
@adamgene can you try this branch again please?